fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -216,7 +216,11 @@ public void alterCatalog(AlterCatalogStmt stmt) {
                     Authorizer.getInstance().setAccessControl(catalogName, new RangerHiveAccessController(serviceName));
                 }
 
-                catalog.getConfig().put("ranger.plugin.hive.service.name", serviceName);
+                catalog.getConfig().putAll(properties);
+                if (serviceName == null && !properties.isEmpty()) {
+                    LOG.info("Altering catalog properties (excluding `ranger.plugin.hive.service.name`) " +
```

Comment:
I believe that the notion of restarting FE to enable newly added properties to take effect is fundamentally not viable.  Could you make this newly added properties taking effect without restarting FE?

Reply:
These properties changes, if applied directly, would affect the in-memory state of the catalog's metadata cache or require re-establishing the connection with the metastore. On one hand, implementing such changes is prone to causing issues and complex, and on the other hand, they would immediately impact business queries with potential irrecoverable consequences. Therefore, i think it is preferable to apply these changes through a restart. This approach simplifies implementation and minimizes the impact on users.

Reply:
I agree that changing the state of the original catalog in place is error-prone. However, I think we can treat the catalog as immutable. Instead of modifying the state of the  original catalog, we can create a new catalog and replace it atomically

Reply:
could we reuse the create catalog logic with alter properties, and repalce the connector object atomically?

Reply:
Solid suggestion. I'll carefully evaluate the feasibility and implications, then circle back with a detailed analysis.

Reply:
@DorianZheng @Youngwb please re review